exp/api/v1beta1/azuremanagedcluster_webhook.go

```diff
@@ -70,6 +71,24 @@ func (r *AzureManagedCluster) ValidateUpdate(oldRaw runtime.Object) error {
 				fmt.Sprintf("annotations with '%s' prefix are immutable", azure.CustomHeaderPrefix)))
 	}
 
+	if old.Spec.ControlPlaneEndpoint.Host != "" {
```

Comment:
Is there a case where `old.Spec.ControlPlaneEndpoint` is nil? I saw that the control plane endpoint is an optional field for the azure managed cluster spec

Reply:
Yes (I assume you mean `""` [empty string], not nil), during initial cluster creation (because this data is never populated by the end user). We can't enforce immutability for transitions like `""` --> `<a valid URL>` because that's the update vector we need to expose to allow the AKS API to provide us with data to do an automated update of this value after the cluster is created (the control plane URL is one piece of output data from the AKS API after a successful cluster create).

Reply:
`ControlPlaneEndpoint` is a plain struct value, not a pointer, so there isn't a risk of dereferencing a nil pointer in this case.